modules/tinymce/src/plugins/link/main/ts/ui/Dialog.ts

```diff
@@ -58,28 +58,28 @@ const collectData = (editor): Promise<LinkDialogInfo> => {
   return DialogInfo.collect(editor, anchorNode);
 };
 
-const getInitialData = (info: LinkDialogInfo, defaultTarget: Option<string>): LinkDialogData => ({
-  url: {
-    value: info.anchor.url.getOr(''),
-    meta: {
-      attach: () => { },
-      text: info.anchor.url.fold(
-        () => '',
-        () => info.anchor.text.getOr('')
-      ),
-      original: {
-        value: info.anchor.url.getOr('')
+const getInitialData = (info: LinkDialogInfo, defaultTarget: Option<string>): LinkDialogData => {
+  const anchor = info.anchor;
+  const url = anchor.url.getOr('');
+
+  return {
+    url: {
+      value: url,
+      meta: {
```

Comment:
I don't quite remember how this data is meant to work for the urlinput, but I assume you've tested that this change doesn't break that functionality?

The old code also looks very odd to me, and now I'm wondering if we have similar anywhere else...

Reply:
Yeah, the data is just used as metadata for the typeahead popup. It's used to match if the URL data has both available, but we only ever store the URL in local storage so it's not meaningful and just causes other issues.

To be honest there's likely a separate issue with the whole Alloy dataset representing logic, but this solves the more immediate issue.